src/v0/sources/shopify/util.js

```diff
@@ -111,11 +120,84 @@ const setAnonymousId = (message) => {
       break;
   }
 };
+/**
+ * This function sets the anonymousId based on cart_token or id from the properties of message.
+ * If it's null then we set userId as "ADMIN".
+ * @param {*} message
+ * @returns
+ */
+const setAnonymousIdorUserIdFromDb = async (message, metricMetadata) => {
+  let cartToken;
+  switch (message.event) {
+    /**
+     * Following events will contain cart_token and we will map it in cartToken
+     */
+    case RUDDER_ECOM_MAP.checkouts_create:
+    case RUDDER_ECOM_MAP.checkouts_update:
+    case SHOPIFY_TRACK_MAP.orders_cancelled:
+    case SHOPIFY_TRACK_MAP.orders_fulfilled:
+    case SHOPIFY_TRACK_MAP.orders_paid:
+    case SHOPIFY_TRACK_MAP.orders_partially_fullfilled:
+    case RUDDER_ECOM_MAP.orders_create:
+    case RUDDER_ECOM_MAP.orders_updated:
+
+      if (!isDefinedAndNotNull(message.properties?.cart_token)) {
+        /**
+         * This case will rise when we will be using Shopify Admin Dashboard to create, update, delete orders etc.
+         * Since it is done by shopify-admin we will set "userId" to be "shopify-admin"
+         */
+        if (!message.userId) {
+          message.setProperty('userId', 'shopify-admin');
+        }
+        return;
+      }
+      cartToken = message.properties?.cart_token;
+      break;
+    /*
+     * we dont have cart_token for carts_create and update events but have id and token field
+     * which later on for orders become cart_token so we are fethcing cartToken from id
+     */
+    case SHOPIFY_TRACK_MAP.carts_create:
+    case SHOPIFY_TRACK_MAP.carts_update:
+      cartToken = message.properties?.id;
+      break;
+    // https://help.shopify.com/en/manual/orders/edit-orders -> order can be edited through shopify-admin only
+    // https://help.shopify.com/en/manual/orders/fulfillment/setting-up-fulfillment -> fullfillments wont include cartToken neither in manual or automatiic
+    case SHOPIFY_TRACK_MAP.orders_edited:
+    case SHOPIFY_TRACK_MAP.fulfillments_create:
+    case SHOPIFY_TRACK_MAP.fulfillments_update:
+      if (!message.userId) {
+        message.setProperty('userId', 'shopify-admin');
+      }
+      return;
+    default:
+  }
+  let anonymousIDfromDB;
+  const executeStartTime = Date.now();
+  const redisVal = await RedisDB.getVal(`${cartToken}`);
+  stats.timing('redis_get_latency', executeStartTime, {
+    ...metricMetadata,
+  });
+  stats.increment('shopify_redis_get_data', {
+    ...metricMetadata,
+    timestamp: Date.now(),
+  });
+  if (redisVal !== null) {
+    anonymousIDfromDB = redisVal.anonymousId;
+  }
+  if (!isDefinedAndNotNull(anonymousIDfromDB)) {
+    /* this is for backward compatability when we don't have the redis mapping for older events
+    we will get anonymousIDFromDb as null so we will set UUID using the session Key */
+    anonymousIDfromDB = cartToken;
```

Comment:
```suggestion
    anonymousIDfromDB = sha256(cartToken).toString().substring(0, 36);
```
For backward compatibility of format